connectors/sources/jira.py

```diff
@@ -390,19 +430,29 @@ async def _get_projects(self):
         try:
             timezone = await self._get_timezone()
 
-            async for response in self.jira_client.api_call(url_name=PROJECT):
-                response = await response.json()
-                for project in response:
-                    yield {
-                        "_id": f"{project['name']}-{project['id']}",
-                        "_timestamp": iso_utc(
-                            when=datetime.now(
-                                pytz.timezone(timezone)  # pyright: ignore
-                            )
-                        ),
-                        "Type": "Project",
-                        "Project": project,
-                    }
+            _timestamp = iso_utc(when=datetime.now(pytz.timezone(timezone)))
+            if self.jira_client.projects == ["*"]:
+                async for response in self.jira_client.api_call(url_name=PROJECT):
+                    response = await response.json()
+                    for project in response:
+                        yield {
+                            "_id": f"{project['name']}-{project['id']}",
+                            "_timestamp": _timestamp,
+                            "Type": "Project",
+                            "Project": project,
+                        }
+            else:
+                for project_key in self.jira_client.projects:
+                    async for response in self.jira_client.api_call(
+                        url_name=PROJECT_BY_KEY, key=project_key
+                    ):
+                        project = await response.json()
+                        yield {
+                            "_id": f"{project['name']}-{project['id']}",
```

Comment:
Only project ID will create conflicts with other Jira objects, so let's update it like `f"jira-project-{project['id']}"`

Reply:
I don't think duplication is going to happen, because if project is renamed then in the next sync, the existing one will be deleted and new project document gets indexed with the new id - `f"{project['name']}-{project['id']}"`.

Reply:
Ok, we can change the scenario - a customer writes a test that checks to make sure that certain IDs are present in their index, and gets frustrated when those "Identifiers" are changing, but the document they care about is still logically the same.

Please change this to not use `name`, which is not a consistent value.

Reply:
Got it! Then let's update it as you suggested i.e. to keep `f"project-{project['id']}"`.